lib/pbench/cli/agent/commands/tools/base.py

```diff
@@ -0,0 +1,62 @@
+import click
+
+from pbench.agent.base import BaseCommand
+
+
+class ListMixIn:
+    """List registered agent tools"""
+
+    def list(self):
+        if self.context.name and self.context.group:
+            click.secho("You cannot specify both --group and --name")
+            return 1
+
+        if not self.pbench_run.exists():
+            self.logger.warn("The %s directory does not exist", self.pbench_run)
+            return 0
+
+        if not self.context.group:
+            # if a group is not specified use all groups
+            self.context.group = self.groups
+
+        if not self.context.name:
+            # list tools in one or all groups
+            dirs = {}
+            for group in self.context.group:
+                _group = group.name.split("tools-v1-")[1]
+                dirs[_group] = {}
+                for path in group.glob("*/**"):
+                    dirs[_group][path.name] = [
+                        t.name
+                        for t in self.tools(path)
+                        if t.name != "__label__" and t.suffix != ".__noinstall__"
```

Comment:
This line break just looks really weird. It'd be easier to read with the `if` indented another level. And, (sigh), if that pushes the line too far out for black, indenting again at the `and`???

Reply:
fixed